### PR TITLE
[SQL] Add managed hsm regex match to SQL

### PR DESCRIPTION
**Pull request by kingsleyAzure**
**Description**  
Extend Current Regex matching to include managed hsm URIs since Managed HSM is supported in SQL now.

### CODE REVIEW

**Comment by yonzhan** [comment]
SQL

**Comment by Juliehzl** [comment]
@bim-msft could you help take a look at hsm part?

**Comment by Juliehzl** [comment]
Hi @kingsleyAzure, could you fix CI failure first?

**Review by bim-msft — APPROVED** [review]
LGTM

**Review by Juliehzl — APPROVED** [review]

**Comment by Juliehzl** [comment]
Hi @kingsleyAzure, could you take a look on style error and make CI pass?

**Comment by kingsleyAzure** [comment]
Hi @Juliehzl 

Can you please retry the pipeline?

**Comment by kingsleyAzure** [comment]
please rerun the pipeline. I fixed the line too long error.

**Comment by Juliehzl** [comment]
Hi @kingsleyAzure, CI is triggered automatically with your commit. But there are more failures occurred. You could see more in `Details`. 
I am fine with your PR, but you need to make CI pass. If you need my help, please ping me in Teams.

File: src/azure-cli/azure/cli/command_modules/rdbms/custom.py [hunk 1]
```diff
@@ -536,11 +536,11 @@ def _get_server_key_name_from_uri(uri):
     '''
     import re
 
-    match = re.match(r'^https(.)+\.vault(.)+\/keys\/[^\/]+\/[0-9a-zA-Z]+$', uri)
-
+    match = re.match(r'https://(.)+\.(managedhsm.azure.net|managedhsm-preview.azure.net|vault.azure.net|vault-int.azure-int.net|vault.azure.cn|managedhsm.azure.cn|vault.usgovcloudapi.net|managedhsm.usgovcloudapi.net|vault.microsoftazure.de|managedhsm.microsoftazure.de|vault.cloudapi.eaglex.ic.gov|vault.cloudapi.microsoft.scloud)(:443)?\/keys/[^\/]+\/[0-9a-zA-Z]+$', uri)
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/custom.py:539]: please add comment to clarify what you do for regex.

Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/custom.py:539]: Do you support Azure Stack cloud? It seems with your current regex, only some known could are supported. If there are new cloud supported, you will need to change the regex here, right?

**Comment by Juliehzl** [comment]
@fengzhou-msft @houk-ms Could you help check the regex for hsm and keyvault?

**Comment by houk-ms** [comment]
@kingsleyAzure May I ask where did you get the list of the KeyVault endpoints? 

I suggest we do some general regex here to match `vault` and `managedhsm` for convenient maintenance, instead of enumerating all the possible host names.

**Comment by kingsleyAzure** [comment]
This was retrieved from endpoint information.

**Comment by kingsleyAzure** [comment]
The same regex match is done in powershell as well.

File: src/azure-cli/azure/cli/command_modules/rdbms/custom.py (hunk 1 above)
Comment (kingsleyAzure) [reply src/azure-cli/azure/cli/command_modules/rdbms/custom.py:539]: currently we do not support Azure stack.

**Review by houk-ms — APPROVED** [review]
LGTM, then.

**Pull request merged**